File: test/java/brs/web/api/http/handler/AbstractTransactionTest.java
package brs.web.api.http.handler;

import brs.Account;
import brs.Attachment;
import brs.SignumException;
import brs.common.AbstractUnitTest;
import brs.web.api.http.common.APITransactionManager;
import com.google.gson.JsonElement;
import org.mockito.ArgumentCaptor;

import javax.servlet.http.HttpServletRequest;

import static org.mockito.ArgumentMatchers.*;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public abstract class AbstractTransactionTest extends AbstractUnitTest {

  @FunctionalInterface
  public interface TransactionCreationFunction<R> {
    R apply() throws SignumException;
  }

  protected Attachment attachmentCreatedTransaction(TransactionCreationFunction r, APITransactionManager apiTransactionManagerMock) throws SignumException {
    final ArgumentCaptor<Attachment> ac = ArgumentCaptor.forClass(Attachment.class);

    when(apiTransactionManagerMock.createTransaction(any(HttpServletRequest.class), nullable(Account.class), nullable(Long.class), anyLong(), ac.capture(), anyLong())).thenReturn(mock(JsonElement.class));

    r.apply();

    return ac.getValue();
  }

}

File: test/java/brs/http/AbstractTransactionTest.kt
package brs.http

import brs.Attachment
import brs.BurstException
import brs.common.AbstractUnitTest
import com.google.gson.JsonElement
import com.google.gson.JsonPrimitive
import com.nhaarman.mockitokotlin2.*

abstract class AbstractTransactionTest : AbstractUnitTest() {
    @Throws(BurstException::class)
    protected fun attachmentCreatedTransaction(r: () -> Any, apiTransactionManagerMock: APITransactionManager): Attachment? {
        val ac = argumentCaptor<Attachment>()

        whenever(apiTransactionManagerMock.createTransaction(any(), any(), anyOrNull(), any(), ac.capture(), any())).thenReturn(JsonPrimitive("hi"))

        r()

        return ac.firstValue
    }

}
